oshi-core/src/main/java/oshi/util/ArchitectureCodename.java

```diff
@@ -0,0 +1,36 @@
+package oshi.util;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+    /**
+     *
+     * Microarchitecture Codename and Processor Model and Family Numbers,
+     * eg. Family 6 Model 94
+     *
+     */
+
+
+public enum ArchitectureCodename {
+
+    /**
+     * Intel
+     */
+
+    SkyLake("6 94");
```

Comment:
This won't work as an enum.  Skylake (Client) has family 6 but models 78 and 94, Skylake (Server) is model 85.

You also may not want to hardcode these values -- that requires releasing OSHI every time a new processor comes out.  I'd suggest a key-value map based on the family/model, using a config file that can be updated by the user if desired.